backend/app/adapter/db/user_changelog_integration_test.go

```diff
@@ -0,0 +1,279 @@
+// +build integration all
+
+package db_test
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/short-d/app/mdtest"
+	"github.com/short-d/short/app/adapter/db"
+	"github.com/short-d/short/app/adapter/db/table"
+	"github.com/short-d/short/app/entity"
+)
+
+var insertUserChangeLogRowSQL = fmt.Sprintf(`
+INSERT INTO %s (%s, %s, %s)
+VALUES ($1, $2, $3)`,
+	table.UserChangeLog.TableName,
+	table.UserChangeLog.ColumnUserID,
+	table.UserChangeLog.ColumnEmail,
+	table.UserChangeLog.ColumnLastViewedAt,
+)
+
+type userChangeLogTableRow struct {
+	userID       string
+	email        string
+	lastViewedAt time.Time
+}
+
+func TestUserChangeLogSQL_GetLastViewedAt(t *testing.T) {
+	now := time.Now()
+	monthAgo := now.AddDate(0, -1, 0)
+
+	testCases := []struct {
+		name                   string
+		userChangeLogTableRows []userChangeLogTableRow
+		user                   entity.User
+		expectedLastViewedAt   time.Time
+		hasErr                 bool
+	}{
+		{
+			name: "user last viewed at time is one month ago",
+			userChangeLogTableRows: []userChangeLogTableRow{
+				{
+					userID:       "12346",
+					email:        "test2@gmail.com",
+					lastViewedAt: now,
+				},
+				{
+					userID:       "12345",
+					email:        "test@gmail.com",
+					lastViewedAt: monthAgo,
+				},
+			},
+			user: entity.User{
+				ID:    "12345",
+				Name:  "Test User",
+				Email: "test@gmail.com",
+			},
+			expectedLastViewedAt: monthAgo,
+			hasErr:               false,
+		},
+		{
+			name: "user does not exist",
+			userChangeLogTableRows: []userChangeLogTableRow{
+				{
+					userID:       "12346",
+					email:        "test2@gmail.com",
+					lastViewedAt: now,
+				},
+			},
+			user: entity.User{
+				ID:    "12345",
+				Name:  "Test User",
+				Email: "test@gmail.com",
+			},
+			expectedLastViewedAt: time.Time{},
+			hasErr:               true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			mdtest.AccessTestDB(
+				dbConnector,
+				dbMigrationTool,
+				dbMigrationRoot,
+				dbConfig,
+				func(sqlDB *sql.DB) {
+					insertUserChangeLogTableRows(t, sqlDB, testCase.userChangeLogTableRows)
+
+					userChangeLogRepo := db.NewUserChangeLogSQL(sqlDB)
+
+					lastViewedAt, err := userChangeLogRepo.GetLastViewedAt(testCase.user)
+					if testCase.hasErr {
+						mdtest.NotEqual(t, nil, err)
+						return
+					}
+
+					mdtest.Equal(t, nil, err)
+					mdtest.Equal(t, testCase.expectedLastViewedAt.Unix(), lastViewedAt.Unix())
```

Comment:
```suggestion
					mdtest.Equal(t, testCase.expectedLastViewedAt.UTC(), lastViewedAt)
```

Reply:
This will not work @byliuyang

Reply:
@alldroll

Reply:
@kapillamba4 Will debug and do experiments on this one.

Reply:
@kapillamba4 How about this: let's add a TODO on top of all `.Unix()` calls and assign the issue to you. I'll let this PR PR after TODO is added.